grails-app/domain/com/unifina/domain/community/CommunitySecret.groovy

```diff
@@ -0,0 +1,35 @@
+package com.unifina.domain.community
+
+import com.unifina.domain.marketplace.Product
+import com.unifina.utils.HexIdGenerator
+import grails.compiler.GrailsCompileStatic
+import groovy.transform.ToString
+
+@ToString
+class CommunitySecret {
+	String id
+	// name to display for users.
+	String name
+	// secret shared by the community that enables automatic join.
+	String secret
+	// communityAddress is an Ethereum address of the community.
+	String communityAddress
+
+    static constraints = {
+		name(nullable: false)
+		secret(nullable: false)
+		communityAddress(nullable: false, validator: Product.isEthereumAddress)
+    }
+	static mapping = {
+		id generator: HexIdGenerator.name
+	}
+
+	@GrailsCompileStatic
+	Map toMap() {
+		return [
+			id: id,
```

Comment:
We probably need to be able to return the `secret` in the object? Especially if it's generated server-side. The API should of course only serve the secrets to the community admin.

Reply:
There is an argument to be made of not providing `secret` in `toMap()`. Having to explicitly put it into an object can prevent accidental revelation of said secret. I could see someone unaware of the details invoking `toMap()` (which sounds innocent enough) and passing the result to some public view.

Reply:
@harbu Sure, but how can a user then get to know what the secret is? (assuming the secret is generated server-side as usual)

I guess this would be in line with the other secrets available from the API.

Reply:
E.g.
```groovy
@StreamrApi
def show() {
   if (userIsAllowedToSeeSecret) {
      def x = communitySecret.toMap()
      x.secret = communitySecret.secret // explicit step
      render(x as JSON)
  }
}
```